mmv1/products/firebaseappcheck/RecaptchaEnterpriseConfig.yaml

```diff
@@ -0,0 +1,81 @@
+# Copyright 2024 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+!ruby/object:Api::Resource
+name: "RecaptchaEnterpriseConfig"
+base_url: projects/{{project}}/apps/{{app_id}}/recaptchaEnterpriseConfig
+self_link: projects/{{project}}/apps/{{app_id}}/recaptchaEnterpriseConfig
+create_url: projects/{{project}}/apps/{{app_id}}/recaptchaEnterpriseConfig?updateMask=tokenTtl,siteKey
+create_verb: :PATCH
+update_verb: :PATCH
+update_mask: true
+skip_delete: true
+description: |
+  An app's reCAPTCHA Enterprise configuration object.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    "Official Documentation": "https://firebase.google.com/docs/app-check"
+  api: "https://firebase.google.com/docs/reference/appcheck/rest/v1/projects.apps.recaptchaEnterpriseConfig"
+import_format:
+  [
+    "projects/{{project}}/apps/{{app_id}}/recaptchaEnterpriseConfig",
+    "{{project}}/{{app_id}}",
+    "{{app_id}}",
+  ]
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "firebase_app_check_recaptcha_enterprise_config_basic"
+    # Need the time_sleep resource
+    pull_external: true
+    primary_resource_id: "default"
+    vars:
+      token_ttl: "7200s"
+      # A valid key, but is actually unusable
+      site_key: "6LdpMXIpAAAAANkwWQPgEdjEhal7ugkH9RK9ytuw"
+    test_vars_overrides:
+      # Don't add random suffix
+      token_ttl: '"7200s"'
+      site_key: '"6LdpMXIpAAAAANkwWQPgEdjEhal7ugkH9RK9ytuw"'
+    test_env_vars:
+      project_id: :PROJECT_NAME
+parameters:
+  - !ruby/object:Api::Type::String
+    name: app_id
+    description: |
+      The ID of an
+      [Web App](https://firebase.google.com/docs/reference/firebase-management/rest/v1beta1/projects.webApps#WebApp.FIELDS.app_id).
+    required: true
+    immutable: true
+    url_param_only: true
+properties:
+  - !ruby/object:Api::Type::String
+    name: name
+    description: |
+      The relative resource name of the reCAPTCHA Enterprise configuration object
+    output: true
+  - !ruby/object:Api::Type::String
+    name: tokenTtl
+    description: |
+      Specifies the duration for which App Check tokens exchanged from reCAPTCHA Enterprise artifacts will be valid.
+      If unset, a default value of 1 hour is assumed. Must be between 30 minutes and 7 days, inclusive.
+
+      A duration in seconds with up to nine fractional digits, ending with 's'. Example: "3.5s".
+    default_from_api: true
```

Comment:
I see this field as part of the creation URL, why is it also default_from_api?

Reply:
The `updateMask` parameter specifies which field to update. The value of the `tokenTtl` itself is passed in the request body. See https://google.aip.dev/161 for more info about field masks in general. In this case, if the request sends an empty value `tokenTtl`, the backend will assign a default that is currently `3600s`, but may change in the future, thus the `default_from_api`.